### PR TITLE
Port this module to metaman's premake system

### PR DESCRIPTION
**Pull request by SwadicalRag**
hon 🥖  hon🥖

### CODE REVIEW

**Comment by meepen** [comment]
🗳

File: projects/premake5.lua [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+		IncludeSDKCommon()
+		IncludeSDKTier0()
+		IncludeSDKTier1()
+		IncludeSDKTier2()
+		IncludeSDKMathlib()
+		-- IncludeSDKRaytrace()
+		IncludeSteamAPI()
+		IncludeDetouring()
+		IncludeScanning()
```
Comment (bmwalters) [inline projects/premake5.lua:35]: Do these need to be included? Ditto with the other target

Comment (SwadicalRag) [reply projects/premake5.lua:35]: you'll need to git clone danielga/sourcesdk-minimal and danielga/garrysmod_common and set their paths to the SOURCE_SDK and GARRYSMOD_COMMON env variables respectively

Comment (SwadicalRag) [reply projects/premake5.lua:35]: those two repos aren't designed to be used as submodules

Comment (bmwalters) [reply projects/premake5.lua:35]: Thanks but I was asking if these libs that you're `Include()`ing are necessary for the module or if they could be commented out to improve binary size.

Comment (SwadicalRag) [reply projects/premake5.lua:35]: OH

not at this stage but I plan on using them in the future so why not include them now

Comment (bmwalters) [reply projects/premake5.lua:35]: Ok, LGTM then

**Review by bmwalters — APPROVED** [review]

File: projects/premake5.lua (hunk 1 above)
Comment (SwadicalRag) [reply projects/premake5.lua:35]: #1 minecraft in gmod uber alles

Comment (MDFL64) [reply projects/premake5.lua:35]: why cant i react with 💯

**Comment by meepdarknessmeep** [comment]
<@140286749996744714> is gay

**Comment by meepdarknessmeep** [comment]
<@151528076247826432> is too

**Comment by meepdarknessmeep** [comment]
<@85855227005128704>

**Comment by meepdarknessmeep** [comment]
<@150809682318065664>

**Comment by meepdarknessmeep** [comment]
<@150809682318065664>

**Comment by MDFL64** [comment]
wut in tarnation

this boi aint raight

**Comment by meepen** [comment]
wat u talkin boot

**Comment by SwadicalRag** [comment]
this boi aint right

**Comment by MDFL64** [comment]
APPROVED

**Comment by SwadicalRag** [comment]
yay

**Pull request merged**